`enforce-canonical-classes` wants to simplify custom `@utility` into itself - creating a circular dependency

### Description

I have a custom utility class:

```css
@utility default-transition {
	@apply transition-all duration-800;
}
```

The `enforce-canonical-classes` rule throws this error:

```
error  The classes: "transition-all, duration-800" can be simplified to "default-transition"  better-tailwindcss/enforce-canonical-classes
```

A fix changes the declaration to:

```css
@utility default-transition {
	@apply default-transition;
}
```

Thus creating a circular dependency.

### Flavor

Vue

### Code Input

```tsx
@utility default-transition {
	@apply transition-all duration-800;
}
```

### Expected Behavior

```tsx
Keep the declaration as is.
```

### Actual Behavior

```tsx
@utility default-transition {
	@apply default-transition;
}
```

### Reproduction URL

_No response_

### ESLint Log

```shell
path\to\file\_transitions.css
   6:9   error  The classes: "transition-all, duration-800" can be simplified to "default-transition"  better-tailwindcss/enforce-canonical-classes
```

### ESLint Config

```typescript
{
			files: ['**/*.css'],
			language: 'css/css',
			languageOptions: {
				customSyntax: tailwind4,
				tolerant: true,
			},
			plugins: {
				css,
				'better-tailwindcss': pluginBetterTailwindcss,
			},
			settings: betterTailwindcssSettings,
		},
```

### Versions

- ESLint: "eslint": "10.3.0",
- Parser: "@eslint/css": "^1.1.0",
- Plugin: "eslint-plugin-better-tailwindcss": "^4.5.0",
- Node: 24

--- a/src/parsers/css.ts
+++ b/src/parsers/css.ts
@@ -50,6 +50,7 @@ function getLiteralsByAtrule(ctx: Rule.RuleContext, node: Atrule): CSSClassListL
   const indentation = getIndentation(line);
   const whitespaces = getWhitespace(content);
   const type = "CSSClassListLiteral";
+  const utility = getEnclosingUtilityName(ctx, node);
 
   return {
     ...whitespaces,
@@ -62,9 +63,42 @@ function getLiteralsByAtrule(ctx: Rule.RuleContext, node: Atrule): CSSClassListL
     raw: content,
     supportsMultiline: true,
     trailingSemicolon,
-    type
+    type,
+    utility
   };
 
+}
+
+function getEnclosingUtilityName(ctx: Rule.RuleContext, node: Atrule): string | undefined {
+  if(isUtilityAtRule(node)){
+    return getUtilityName(ctx, node);
+  }
+
+  // @ts-expect-error - CSS Tree types are different
+  const ancestors = ctx.sourceCode.getAncestors(node);
+
+  for(const ancestor of ancestors){
+    if(isAtRule(ancestor) && isUtilityAtRule(ancestor)){
+      return getUtilityName(ctx, ancestor);
+    }
+  }
+}
+
+function getUtilityName(ctx: Rule.RuleContext, node: Atrule): string | undefined {
+  if(isUtilityAtRule(node)){
+    return node.prelude?.type === "AtrulePrelude" || node.prelude?.type === "Raw"
+      // @ts-expect-error - CSS Tree types are different
+      ? ctx.sourceCode.getText(node.prelude)
+      : undefined;
+  }
+}
+
+function isUtilityAtRule(node: Atrule): boolean {
+  return node.name === "utility";
+}
+
+function isAtRule(node: any): node is Atrule {
+  return node.type === "Atrule";
 }
 
 function getLoc(ctx: Rule.RuleContext, node: Atrule, startOffset: number, endOffset: number): Loc["loc"] | undefined {

--- a/src/rules/enforce-canonical-classes.test.ts
+++ b/src/rules/enforce-canonical-classes.test.ts
@@ -87,6 +87,77 @@ describe.runIf(getTailwindCSSVersion().major >= 4)(enforceCanonicalClasses.name,
     });
   });
 
+  it("should not suggest a canonical class when it matches the enclosing utility name", () => {
+    lint(enforceCanonicalClasses, {
+      valid: [
+        {
+          css: css`
+            @utility default-transition {
+              @apply transition-all duration-800;
+            }
+          `,
+
+          files: {
+            "styles.css": css`
+              @import "tailwindcss";
+
+              @utility default-transition {
+                @apply transition-all duration-800;
+              }
+            `
+          },
+          options: [{
+            entryPoint: "styles.css"
+          }]
+        }
+      ]
+    });
+  });
+
+  it("should still suggest canonical classes inside utility blocks when output differs", () => {
+    lint(enforceCanonicalClasses, {
+      invalid: [
+        {
+          css: css`
+            @utility not-size {
+              @apply w-10 h-10;
+            }
+          `,
+          cssOutput: css`
+            @utility not-size {
+              @apply size-10 ;
+            }
+          `,
+
+          errors: [
+            {
+              message: values(enforceCanonicalClasses.messages!.multiple, {
+                canonicalClass: "size-10",
+                classNames: "w-10, h-10"
+              })
+            },
+            {
+              message: values(enforceCanonicalClasses.messages!.multiple, {
+                canonicalClass: "size-10",
+                classNames: "w-10, h-10"
+              })
+            }
+          ],
+
+          files: {
+            "styles.css": css`
+              @import "tailwindcss";
+            `
+          },
+          options: [{
+            collapse: true,
+            entryPoint: "styles.css"
+          }]
+        }
+      ]
+    });
+  });
+
   it("should collapse multiple utilities into a single utility", () => {
     lint(enforceCanonicalClasses, {
       invalid: [

--- a/src/rules/enforce-canonical-classes.ts
+++ b/src/rules/enforce-canonical-classes.ts
@@ -93,6 +93,13 @@ function lintLiterals(ctx: Context<typeof enforceCanonicalClasses>, literals: Li
         return;
       }
 
+      if(
+        literal.utility &&
+        canonicalClass.output === literal.utility
+      ){
+        return;
+      }
+
       if(canonicalClass.input.length > 1){
         return {
           data: {

--- a/src/types/ast.ts
+++ b/src/types/ast.ts
@@ -71,6 +71,7 @@ export interface StringLiteral extends LiteralBase {
 
 export interface CSSClassListLiteral extends LiteralBase {
   type: "CSSClassListLiteral";
+  utility?: string | undefined;
 }
 
 export type Literal = CSSClassListLiteral | StringLiteral | TemplateLiteral;

